pkg/devfile/devfile.go

```diff
@@ -76,7 +76,14 @@ func GetResourceFromDevfile(log logr.Logger, devfileData data.DevfileData, deplo
 	k8sLabels := generateK8sLabels(compName, appName)
 	matchLabels := getMatchLabel(compName)
 
+	if len(kubernetesComponents) == 0 {
+		return parser.KubernetesResources{}, fmt.Errorf("the devfile has no kubernetes components defined, missing outerloop definition")
```

Comment:
I can't remember what was discussed around this.

What happens when this error is returned? If no Kubernetes components are defined, I feel like we should fall back on gitops resource generation from within HAS? Rather than erroring out

Reply:
it will error out as it's not a valid outerloop devfile. and this error won't be triggered for a normal end-to-end use case. as this error will be detected and reported by the CDQ (change in https://github.com/redhat-appstudio/application-service/pull/278).